app/src/main/java/com/cornellappdev/transit/ui/screens/HomeScreen.kt

```diff
@@ -111,6 +113,14 @@ fun HomeScreen(
         }
     }
 
+    val context = LocalContext.current
+
+    if (permissionState.status.isGranted) {
+        homeViewModel.instantiateLocation(context)
+    }
```

Comment:
I'm not exactly sure if this is the best way to call this? If we only want to call this *once*, then we can use `LaunchedEffect(null) { ... }` to run it only the first time the composable is displayed (and ignore it on recompositions). But I guess if we want to run it on every recomposition for some reason, perhaps it'll work fine.

Reply:
I believe this is what Uplift is doing. If there aren't permissions yet we need to try and enter the if statement on every recomposition until there is, almost like polling for location permissions. I'll make note of whether we experience any issues with actually calling instantiateLocation multiple times.